pkg/controllers/controllers.go

```diff
@@ -67,6 +67,7 @@ func NewControllers(
 	recorder events.Recorder,
 	cloudProvider cloudprovider.CloudProvider,
 	cluster *state.Cluster,
+	wellKnownLabels map[string]string,
```

Comment:
Rather than injecting the well known labels **specifically** for the metrics controller here, I think it would make more sense to just inject them as part of the cloudprovider interface and **never** rely on them through global injection. That way, we aren't relying on two sources of truth for the same labels. Minimally, if we want to inject this here I'd just fame it as "additional node metric labels" rather than well-known labels.

Reply:
+1 on this -- it would be nice if we could just get all the well-known labels from just calling GetInstanceTypes() and collecting all of the labels from the instance type requirements and offering requirements so we wouldn't have to create a separate method -- though, I don't know if that's really possible with things like capacity-reservation-id and it might lead to more misleading bugs in general -- but something to think about

Reply:
We won't need to do this anymore, since will be relaying on runtime metrics creation.